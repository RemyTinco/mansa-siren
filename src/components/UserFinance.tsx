import React, { useEffect, useState } from 'react';
import mansaAPI from 'api/mansa';
import _ from 'lodash';
import { TAccount } from 'types';
import Account from 'components/Account';
//import AccountChart from 'components/chart/AccountChart';
import Modal from 'components/chart/Modal';

const UserFinance = (): JSX.Element => {
  //State to stock a fetched data
  const [accountsData, setAccountsData] = useState<TAccount[]>([]);

  //State to handle a chosen account
  const [currentAccount, setCurrentAccount] = useState<string>('');

  //State to handle an error
  const [error, setError] = useState<string>('');

  //State to handle a chosen account
  const [openModal, setOpenModal] = useState<boolean>(false);

  useEffect(() => {
    (async function getBusinessData() {
      try {
        const response = await mansaAPI.get(`/accounts`);

        // Picking up just an appropriate data
        const fetchedData: TAccount[] = response.data.map((account) => {
          return _.pick(account, [
            'account_id',
            'account_type',
            'account_number',
            'currency',
            'current',
          ]);
        });

        setAccountsData(fetchedData);
      } catch (e) {
        setError(e.toString());
      }
    })();
  }, []); // Doing just at once

  const onAccountSelect = (id: string): void => {
    setCurrentAccount(id);
  };

  const onOpenModal = (status: boolean): void => {
    setOpenModal(status);
  };

  // Helping function to generate accounts
  const renderAccounts = (accountsData: TAccount[]): JSX.Element[] => {
    return accountsData.map(
      (account: TAccount): JSX.Element => {
        return (
          <Account
            {...account}
            key={account.account_id}
            onAccountSelect={onAccountSelect}
            onOpenModal={onOpenModal}
          />
        );
      }
    );
  };

  // Shows up an error if exists
  if (error) {
    return <div className='ui header red'>{error}</div>;
  }
  // Loading while the accountsData is an empty array
  if (accountsData.length === 0) {
    return <div className='ui active centered inline loader big'></div>;
  }

  return (
    <>
      <div className='ui link cards centered'>
        {renderAccounts(accountsData)}
      </div>
      <Modal
        openModal={openModal}
        onDismiss={() => onOpenModal(false)}
        currentAccount={currentAccount}
      />
    </>
  );
};

export default UserFinance;
