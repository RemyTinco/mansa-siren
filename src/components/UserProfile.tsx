import React from 'react';
import faker from 'faker';
import withStyles, { WithStylesProps } from 'react-jss';
import UserBusiness from 'components/UserBusiness';

const styles = {
  profile: {
    minWidth: 400,
    marginBottom: '40px!important',
  },
};

interface UserProfileProps extends WithStylesProps<typeof styles> {}

const UserProfile: React.FunctionComponent<UserProfileProps> = ({
  classes,
}) => {
  return (
    <div className={`ui card centered ${classes.profile}`}>
      <div className='content header'>
        <img
          className='ui avatar image bordered floated right'
          src={faker.image.avatar()}
          alt='avatar'
        />
        <strong>
          <span>{faker.name.firstName()}&nbsp;</span>
          <span>{faker.name.lastName()}</span>
        </strong>
        <div className='meta'>
          <span className='date'>{faker.name.jobTitle()}</span>
        </div>
      </div>

      <div className='content'>
        <h4 className='ui sub header'>Business information</h4>
        <UserBusiness />
      </div>
      <div className='extra content'>
        <i className='phone icon'></i>
        {faker.phone.phoneNumber()}
      </div>
    </div>
  );
};

export default withStyles(styles)(UserProfile);
