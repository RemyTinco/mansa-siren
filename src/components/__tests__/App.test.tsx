import React from 'react';
import { mount } from 'enzyme';
import App from 'components/App';
import UserProfile from 'components/UserProfile';
import UserFinance from 'components/UserFinance';

let wrapper;

describe('App component', () => {
  beforeEach(() => {
    wrapper = mount(<App />);
  });

  afterEach(() => {
    wrapper.unmount();
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('shows the user profile block', () => {
    expect(wrapper.find(UserProfile).length).toEqual(1);
  });

  it('shows the user financial block', () => {
    expect(wrapper.find(UserFinance).length).toEqual(1);
  });
});
