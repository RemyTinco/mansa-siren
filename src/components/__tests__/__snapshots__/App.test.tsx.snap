// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<JssContextSubscriber(App)>
  <WithStyles(App)
    innerRef={null}
    jssContext={
      Object {
        "classNamePrefix": "",
        "disableStylesGeneration": false,
      }
    }
    theme={Object {}}
  >
    <App
      classes={
        Object {
          "wrapper": "App-wrapper-0-2-1",
        }
      }
    >
      <div
        className="App-wrapper-0-2-1"
      >
        <JssContextSubscriber(UserProfile)>
          <WithStyles(UserProfile)
            innerRef={null}
            jssContext={
              Object {
                "classNamePrefix": "",
                "disableStylesGeneration": false,
              }
            }
            theme={Object {}}
          >
            <UserProfile
              classes={
                Object {
                  "profile": "UserProfile-profile-0-2-2",
                }
              }
            >
              <div
                className="ui card centered UserProfile-profile-0-2-2"
              >
                <div
                  className="content header"
                >
                  <img
                    alt="avatar"
                    className="ui avatar image bordered floated right"
                    src="https://s3.amazonaws.com/uifaces/faces/twitter/drewbyreese/128.jpg"
                  />
                  <strong>
                    <span
                      className="firstName"
                    >
                      Laurine
                      Â 
                    </span>
                    <span
                      className="lastName"
                    >
                      Schulist
                    </span>
                  </strong>
                  <div
                    className="meta"
                  >
                    <span
                      className="date jobTitle"
                    >
                      Legacy Factors Developer
                    </span>
                  </div>
                </div>
                <div
                  className="content"
                >
                  <h4
                    className="ui sub header"
                  >
                    Business information
                  </h4>
                  <UserBusiness>
                    <div
                      className="ui active centered inline loader"
                    />
                  </UserBusiness>
                </div>
                <div
                  className="extra content"
                >
                  <i
                    className="phone icon"
                  />
                  <span
                    className="phoneNumber"
                  >
                     
                    955.199.5876 x161
                  </span>
                </div>
              </div>
            </UserProfile>
          </WithStyles(UserProfile)>
        </JssContextSubscriber(UserProfile)>
        <UserFinance>
          <div
            className="ui active centered inline loader big"
          />
        </UserFinance>
      </div>
    </App>
  </WithStyles(App)>
</JssContextSubscriber(App)>
`;
